packages/flutter/lib/src/services/system_channels.dart

```diff
@@ -218,7 +217,7 @@ class SystemChannels {
   ///
   ///  * [WidgetsBindingObserver.didChangeAppLifecycleState], which triggers
   ///    whenever a message is received on this channel.
-  static const BasicMessageChannel<String> lifecycle = BasicMessageChannel<String>(
+  static const BasicMessageChannel<String?> lifecycle = BasicMessageChannel<String?>(
```

Comment:
This surprises me, why is null allowed here? (according to the doc it only sends string representation of enum values.)

Reply:
Because `StringCodec` implements `MessageCodec<String?>` and its decode/encode methods accepts/returns null.